Lib/pydoc.py

```diff
@@ -225,6 +225,19 @@ def classname(object, modname):
         name = object.__module__ + '.' + name
     return name
 
+def parentname(object, modname):
+    """Get a name of the enclosing class (qualified it with a module name
+    if necessary) or module."""
+    if '.' in object.__qualname__:
+        name = object.__qualname__.rpartition('.')[0]
+        if object.__module__ != modname:
+            return object.__module__ + '.' + name
+        else:
+            return name
+    else:
+        if object.__module__ != modname:
+            return object.__module__
```

Comment:
```suggestion
    elif object.__module__ != modname:
         return object.__module__
```

This function can also return `None`, not sure if this is desired 🤔 

In that case it would render as `from None` in https://github.com/python/cpython/pull/113941/files#diff-d2599cb4ccbf37634c5d3089a2750fc6c92c6c98d2a65861ff13d08ed3e9cff5R713-R714

Reply:
In the current form both branches of `if '.' in object.__qualname__:` are more symmetric.

And this function always returns non-None if link is not-None.